third_party/hbase-spark-connector/hbase-connectors/src/main/scala/com/google/cloud/spark/bigtable/ReadRowConversions.scala

```diff
@@ -63,17 +64,27 @@ object ReadRowConversions extends Serializable {
       fields: Seq[Field],
       bigtableRow: BigtableRow,
       catalog: BigtableTableCatalog
-  ): SparkRow = {
+  ): Option[SparkRow] = {
     val keySeq = catalog.row.parseFieldsFromBtRow(bigtableRow)
 
     val valueSeq = fields
       .filter(!_.isRowKey)
+      .filter(field => !catalog.sMap.cqMap.contains(field.sparkColName))
       .map(field => extractValue(field, bigtableRow))
-      .toMap
     val cqFields = catalog.sMap.cqMap.values.toSeq
-    val cqValueSeq = buildRowForRegex(cqFields, bigtableRow).toMap
-    val unionedRow = keySeq ++ valueSeq ++ cqValueSeq
-    SparkRow.fromSeq((fields ++ cqFields).map(unionedRow.get(_).orNull))
+    val cqValueSeq = buildRowForRegex(cqFields, bigtableRow)
+
+    if (catalog.sMap.cqMap.nonEmpty) {
+      val nonNullValues = valueSeq.filter(_._2 != null)
+      val nonEmptyMaps = cqValueSeq.filter(!_._2.isEmpty)
+
+      if (nonNullValues.isEmpty && nonEmptyMaps.isEmpty) {
+        return None
```

Comment:
do we need to update the method that calls `buildRow`? what happens now if `buildRow` returns None?

Reply:
I already updated the methods that called buildRow with `flatMap`, which filter out the None elements